prow/io/opener.go

```diff
@@ -93,6 +98,23 @@ func NewOpener(ctx context.Context, gcsCredentialsFile, s3CredentialsFile string
 	}, nil
 }
 
+func createGCSClient(ctx context.Context, gcsCredentialsFile string) (storageClient, error) {
```

Comment:
@fejta @alvaroaleman So I think that should implement what you suggested. If that's alright, the only question is what to do with the `gcsNoAuth` flag. Should I just ignore it, deprecate it and add a deprecation warning if its set?

Reply:
implementation looks good. For the flag, can you just make it a no-op and emit a deprecation warning when set? We can then remove it in a couple of weeks

Reply:
Done